components/feature/tabs/src/main/java/mozilla/components/feature/tabs/CustomTabsUseCases.kt

```diff
@@ -37,7 +37,7 @@ class CustomTabsUseCases(
             customTabConfig: CustomTabConfig,
             private: Boolean = false,
             additionalHeaders: Map<String, String>? = null,
-            source: SessionState.Source = SessionState.Source.CUSTOM_TAB
+            source: SessionState.Source
```

Comment:
nit: Hm should we default to something here? Kind of strange that we allow a `CustomTabUseCase` to be invoked with e.g. SessionState.Source.ActionSearch` for instance.

Reply:
Maybe... the previous behaviour was basically off in the same way. The default helps but doesn't solve the problem, which is that this isn't typed correctly because our types are somewhat misaligned.

e.g. ideally this is something like `source: SessionState.Source.CustomTab`, or perhaps just `SessionState.CustomTab`. However, the internal/external split doesn't allow doing this since Kotlin doesn't have union types.

I suppose moving `CustomTab` into its own source "subtype" is one approach here to improve the situation:
```
Source {
  CustomTab: {Internal, External(caller) },
  Internal: {...}, External: {...}
}
```

... then for CustomTabSessionState and wherever we have a CustomTab being created, we can properly type things - `source: SessionState.Source.CustomTab`.

Not sure this is worth doing before landing though, but i'll leave a note in SessionState about it.

Reply:
Yeah, we don't have to handle this now. Just a though I had when looking at the API, might not be worth fixing.